templates/internal/appName/http/httpservice.go.tpl

```diff
@@ -48,16 +56,37 @@ func (s *HTTPService) Run(ctx context.Context) error {
 }
 
 {{- if has "http" (stencil.Arg "serviceActivities") }}
+// PublicHTTPDependencies is used to inject dependencies into the PublicHTTPService
+// service activity.
+type PublicHTTPDependencies struct{
+    // <<Stencil::Block(publicHTTPDependencies)>>
+{{ file.Block "publicHTTPDependencies" }}
+	  // <</Stencil::Block>>
+
+    {{- $publicHTTPDependencyInjection := stencil.GetModuleHook "internal/http/publicHTTPDependencyInjection" }}
+    {{- if $publicHTTPDependencyInjection }}
+    // dependencies injected by modules
+    {{- range $publicHTTPDependencyInjection }}
+    {{ . }}
+    {{- end }}
+    // end dependencies injected by modules
+    {{- end }}
+}
+
 // PublicHTTPService handles public http service calls
 type PublicHTTPService struct {
-	handlers.PublicService
+	  handlers.PublicService
 
-	cfg *Config
+	  cfg *Config
+    deps *PublicHTTPDependencies
 }
 
 // NewPublicHTTPService creates a new PublicHTTPService service activity.
-func NewPublicHTTPService(cfg *Config) *PublicHTTPService {
-	return &PublicHTTPService{cfg: cfg}
+func NewPublicHTTPService(cfg *Config, deps *PublicHTTPDependencies) *PublicHTTPService {
```

Comment:
Can you mark the old signature deprecated, and create a new one instead? I'd like to start giving customers some advance notice for breaking changes.

Ditto for the `StartServers` call, which [is being used](https://cs.github.com/getoutreach/kaiagateway/blob/ad1f20dbf839c9b6f346632617e2f43da7926b2b/internal/kaiagateway/bento_gateway_test.go?q=startservers+#L39).

Reply:
Not doing this as per the conversation we had about limited blast radius here.